pkg/controller/direct/sql/sqlinstance_mappings.go

```diff
@@ -30,12 +30,6 @@ func SQLInstanceKRMToGCP(in *krm.SQLInstance) (*api.DatabaseInstance, error) {
 		return nil, fmt.Errorf("cannot convert nil KRM SQLInstance to GCP DatabaseInstance")
 	}
 
-	if in.Spec.CloneSource != nil {
```

Comment:
Could you share some context why this check is not needed?

Reply:
From the commit message:
> Direct controller needs to be able to update SQLInstance with cloneSource specified.